pkg/session/session.go

```diff
@@ -30,88 +31,152 @@ type session struct {
 	isAuthenticated bool
 }
 
-func (s session) Info() string {
+func (s *session) Info() string {
 	return fmt.Sprintf("Brokers: %v, Authenticated: %v", s.brokers, s.isAuthenticated)
 }
 
-func (s session) Connect(ctx context.Context) (sarama.Client, error) {
+func (s *session) Connect(ctx context.Context) (sarama.Client, error) {
 	if s.client != nil {
 		return s.client, nil
 	}
-
 	client, err := cluster.NewCluster(s.brokers).Connect(ctx)
 	if err != nil {
 		return nil, fmt.Errorf("error connecting to cluster: %w", err)
 	}
-
 	adminClient, err := cluster.NewCluster(s.brokers).ConnectAdmin(ctx)
-
 	if err != nil {
 		return nil, fmt.Errorf("error connecting to cluster as admin: %w", err)
 	}
-
 	s.client = client
 	s.adminClient = adminClient
 	s.isAuthenticated = true
-	currentSession = s
 	return client, nil
 }
 
-func (s session) Disconnect() {
+func (s *session) Disconnect() {
 	if s.client != nil {
 		s.client.Close()
 	}
 	s.client = nil
 	s.adminClient = nil
 	s.isAuthenticated = false
-	currentSession = s
 	fmt.Println("Logged out successfully!")
 }
 
-func (s session) IsAuthenticated() bool {
+func (s *session) IsAuthenticated() bool {
 	return s.isAuthenticated
 }
 
-func (s session) GetClient() sarama.Client {
+func (s *session) GetClient() sarama.Client {
 	if s.client == nil {
 		return nil
 	}
 	return s.client
 }
 
-func (s session) GetAdminClient() sarama.ClusterAdmin {
+func (s *session) GetAdminClient() sarama.ClusterAdmin {
 	if s.adminClient == nil {
 		return nil
 	}
 
 	return s.adminClient
 }
 
-func GetCurrentSession() Session {
+func GetCurrentSession() *session {
 	return currentSession
 }
 
-var currentSession Session
+var currentSession *session
+
+func createDefaultSession() error {
+	file, err := os.Create(".openkommander_config")
+	if err != nil {
+		fmt.Println("Error creating session file:", err)
+		return err
+	}
+
+	defer file.Close()
+
+	sessionData := map[string]interface{}{
```

Comment:
using a struct for the session data will make the code more readable, you can pass a struct to the encoder/decoder and if the fields are tagged correctly, it will work out of the box:

```
type SessionData struct {
  Brokers []string `json:"brokers"`
  IsAuthenticated bool `json:"isAuthenticated"`
}

...

session := SessionData{ isAuthenticated: false }
encoder.Encode(session)

```